apollo-router/src/lib.rs

```diff
@@ -26,19 +25,40 @@ use displaydoc::Display as DisplayDoc;
 use futures::channel::{mpsc, oneshot};
 use futures::prelude::*;
 use futures::FutureExt;
+use once_cell::sync::Lazy;
 use once_cell::sync::OnceCell;
 use std::path::{Path, PathBuf};
 use std::pin::Pin;
+use std::sync::{Arc, Mutex};
 use std::task::{Context, Poll};
 use std::time::Duration;
 use thiserror::Error;
 use tokio::task::spawn;
+use tracing::Dispatch;
 use Event::{Shutdown, UpdateConfiguration, UpdateSchema};
 
 type SchemaStream = Pin<Box<dyn Stream<Item = graphql::Schema> + Send>>;
 
 pub static GLOBAL_ENV_FILTER: OnceCell<String> = OnceCell::new();
 
+static SUBSCRIBER: Lazy<Mutex<Option<Arc<dyn tracing::Subscriber + Send + Sync + 'static>>>> =
+    Lazy::new(|| Mutex::new(None));
+
+/// Retrieve a new dispatcher which uses the global subscriber
+pub fn get_dispatcher() -> Dispatch {
+    let sub_guard = SUBSCRIBER.lock().unwrap();
+
+    sub_guard
+        .clone()
+        .map(tracing::Dispatch::new)
+        .unwrap_or_default()
```

Comment:
If there is a default dispatcher why not directly put a `Dispatch::default()` instead of an option as we call `unwrap_or_default()` ?

Reply:
In order to do what you are suggesting, I'd have to replace the stored Subscriber (which needs to be optional) with a stored Dispatch. I'm probably ok with doing that, but that's an "extra" change above what I was trying to do to just migrate what we have into a plugin.

Reply:
Ok let's keep the current implementation then